ghost/core/core/server/services/themes/validate.js

```diff
@@ -41,16 +60,57 @@ const check = async function check(theme, isZip) {
     }
 
     checkedTheme = gscan.format(checkedTheme, {
-        onlyFatalErrors: config.get('env') === 'production',
+        onlyFatalErrors: false,
         checkVersion: checkedVersion
     });
 
+    // In production we don't want to show warnings
+    // Warnings are meant for developers only
+    if (config.get('env') === 'production') {
+        checkedTheme.results.warning = [];
+    }
+
+    // Cache the result
+    try {
+        await gscanCacheStore.set(themeName, checkedTheme);
```

Comment:
Have you looked into the size of the "checkedTheme" object? I do remember we shove quite a lot of information there, not too crazy but there might be a bit of a memory overhead added to what the instance normally uses. Did you run napkin math of how much that would be just to be sure nothing horrific is happening?